source/common/http/conn_manager_impl.cc

```diff
@@ -1879,6 +1872,26 @@ void ConnectionManagerImpl::ActiveStream::onResetStream(StreamResetReason, absl:
   ENVOY_STREAM_LOG(debug, "stream reset", *this);
   connection_manager_.stats_.named_.downstream_rq_rx_reset_.inc();
   connection_manager_.doDeferredStreamDestroy(*this);
+
+  // We interpret a LocalReset as a downstream protocol error. This
+  // interpretation appears to be correct in today's code
+  // (2020-02-10).
+  if (reason == StreamResetReason::LocalReset) {
+    stream_info_.setResponseFlag(StreamInfo::ResponseFlag::DownstreamProtocolError);
```

Comment:
I'm not sure this is correct.
I think in the case there's an error and envoy does sendLocalResponse, maybeEndEncode() will call doEndStream
to indicate that the stream is fully ended (we don't allow continued POST body upload if we're sending a local response) doEndStream will resetStream also with LocalReset. This path can be triggered by other errors, such as unexpected upstream disconnect.